integration-core/src/main/java/org/symphonyoss/integration/core/service/CryptoServiceImpl.java

```diff
@@ -0,0 +1,204 @@
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implementation of a text-based cryptography service.
+ * Created by campidelli on 9/5/17.
+ */
+@Component
+public class CryptoServiceImpl implements CryptoService {
+
+  private static final int ITERATIONS = 65536;
+  private static final int KEY_SIZE = 256;
+  private static final int SALT_SIZE = 20;
+  private static final String KEY_ALGORITHM = "PBKDF2WithHmacSHA1";
+  private static final String SPEC_ALGORITHM = "AES";
+  private static final String MODE = "CBC";
+  private static final String PADDING = "PKCS5Padding";
+  private static final String TRANSFORMATION = SPEC_ALGORITHM + "/" + MODE + "/" + PADDING;
+  private static final String CHARSET = "UTF-8";
+
+  private static final String INVALID_PARAMETER = "core.crypto.invalid.parameter";
+  private static final String INVALID_PARAMETER_SOLUTION = INVALID_PARAMETER + ".solution";
+  private static final String NO_SUCH_ALGORITHM = "core.crypto.no.such.algorithm";
+  private static final String NO_SUCH_ALGORITHM_SOLUTION = NO_SUCH_ALGORITHM + ".solution";
+  private static final String INVALID_KEY_SPEC = "core.crypto.invalid.key.spec";
+  private static final String INVALID_KEY_SPEC_SOLUTION = INVALID_KEY_SPEC + ".solution";
+  private static final String NO_SUCH_PADDING = "core.crypto.no.such.padding";
+  private static final String NO_SUCH_PADDING_SOLUTION = NO_SUCH_PADDING + ".solution";
+  private static final String INVALID_KEY = "core.crypto.invalid.key";
+  private static final String INVALID_KEY_SOLUTION = INVALID_KEY + ".solution";
+  private static final String INVALID_PARAM_SPEC = "core.crypto.invalid.param.spec";
+  private static final String INVALID_PARAM_SPEC_SOLUTION = INVALID_PARAM_SPEC + ".solution";
+  private static final String ILLEGAL_BLOCK_SIZE = "core.crypto.illegal.block.size";
+  private static final String ILLEGAL_BLOCK_SIZE_SOLUTION = ILLEGAL_BLOCK_SIZE + ".solution";
+  private static final String BAD_PADDING = "core.crypto.bad.padding";
+  private static final String BAD_PADDING_SOLUTION = BAD_PADDING + ".solution";
+  private static final String UNSUPPORTED_ENCODING = "core.crypto.unsupported.encoding";
+  private static final String UNSUPPORTED_ENCODING_SOLUTION = UNSUPPORTED_ENCODING + ".solution";
+
+  @Autowired
+  private LogMessageSource logMessage;
+
+  /**
+   * @see CryptoService#encrypt(String, String)
+   */
+  @Override
+  public String encrypt(String plainText, String key) throws CryptoException {
+    checkParameters("plainText", plainText);
+    checkParameters("key", key);
+    int blockSize = 0;
+    try {
+      byte[] saltBytes = generateSalt();
+      SecretKeySpec secret = deriveKey(key, saltBytes);
+
+      Cipher cipher = getCipher();
+      initCipher(cipher, Cipher.ENCRYPT_MODE, secret, null);
+      blockSize = cipher.getBlockSize();
```

Comment:
I don't see that it's being used anywhere below. Why is it retrieved?

Reply:
Sure, I forgot it there. Done!